components/custom_status/custom_status_modal.tsx

```diff
@@ -0,0 +1,284 @@
+];
+
+const CustomStatusModal: React.FC<Props> = (props: Props) => {
+    const getCustomStatus = makeGetCustomStatus();
+    const dispatch = useDispatch();
+    const currentCustomStatus = useSelector((state: GlobalState) => getCustomStatus(state)) || {};
+    const recentCustomStatuses = useSelector((state: GlobalState) => getRecentCustomStatuses(state));
+    const customStatusControlRef = useRef(null);
+    const {formatMessage} = useIntl();
+    const [showEmojiPicker, setShowEmojiPicker] = useState<boolean>(false);
+    const [text, setText] = useState<string>(currentCustomStatus.text);
+    const [emoji, setEmoji] = useState<string>(currentCustomStatus.emoji);
+    const isStatusSet = emoji || text;
+    const firstTimeModalOpened = useSelector((state: GlobalState) => showStatusDropdownPulsatingDot(state));
+
+    const handleCustomStatusInitializationState = () => {
+        if (firstTimeModalOpened) {
+            dispatch(setCustomStatusInitialisationState(Preferences.CUSTOM_STATUS_MODAL_VIEWED));
+        }
+    };
+
+    useEffect(handleCustomStatusInitializationState, []);
+
+    const handleSetStatus = () => {
+        const customStatus = {
+            emoji: emoji || 'speech_balloon',
+            text: text.trim(),
+        };
+        dispatch(setCustomStatus(customStatus));
+    };
+
+    const handleClearStatus = () => {
+        if (currentCustomStatus.text || currentCustomStatus.emoji) {
+            dispatch(unsetCustomStatus());
+        }
+    };
+
+    const getCustomStatusControlRef = () => {
+        return customStatusControlRef.current;
+    };
+
+    const handleEmojiClose = () => {
+        setShowEmojiPicker(false);
+    };
+
+    const handleEmojiClick = (selectedEmoji: any) => {
+        setShowEmojiPicker(false);
+        const emojiName = selectedEmoji.name || selectedEmoji.aliases[0];
+        setEmoji(emojiName);
+    };
+
+    const toggleEmojiPicker = () => {
+        setShowEmojiPicker(!showEmojiPicker);
+    };
+
+    const handleTextChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const inputText = event.target.value;
+        if (inputText.length <= Constants.CUSTOM_STATUS_TEXT_CHARACTER_LIMIT) {
+            setText(inputText);
+        }
+    };
+
+    const handleRecentCustomStatusClear = (status: any) => {
+        dispatch(removeRecentCustomStatus(status));
+    };
+
+    const customStatusEmoji = emoji || text ?
+        (
+            <RenderEmoji
+                emojiName={emoji || 'speech_balloon'}
+                size={20}
+            />
+        ) : (
+            <EmojiIcon className={'icon icon--emoji'}/>
+        );
+
+    const clearHandle = () => {
+        setText('');
+        setEmoji('');
+    };
+
+    const disableSetStatus = (currentCustomStatus.text === text && currentCustomStatus.emoji === emoji) ||
+        (text === '' && emoji === '');
+
+    const handleSuggestionClick = (status: any) => {
```

Comment:
do we really need an `any` here?
IMO we should refrain from using `any` if possible.

Maybe we can create a type `CustomStatusOption`?

```
type CustomStatusOption = {
  emoji: string;
  text: string;
}
```